src/Components/QuestionList/QuestionList.tsx

```diff
@@ -19,30 +19,59 @@ export const QuestionList = ({ setShowQuestionList }: any) => {
         StorageKey.filteredQuestionBank,
         []
     );
-
     const [_, setCurrentQuestion] = useLocalStorage(
         StorageKey.currentQuestion,
         {}
     );
 
-    const [questionType, setQuestionType] = useState<string>("Js");
+    const [filterSettings, setFilterSettings] = useState<FilterSetting>({
+        type: "Js",
+        name: "",
+        hideCorrect: false,
+        difficulty: "null",
+    });
+
+    useEffect(() => {
+        const { name, type, hideCorrect, difficulty } = filterSettings;
+        let filter = questions;
+
+        if (type) {
+            filter = filter.filter((question: Question, index: number) => {
+                return question.tags.includes(type);
+            });
+        }
+
+        if (name) {
+            filter = filter.filter((question: Question, index: number) => {
+                return question.answer
+                    .toLowerCase()
+                    .includes(name.toLocaleLowerCase());
+            });
+        }
+
+        if (difficulty) {
+            filter = filter.filter((question: Question, index: number) => {
+                return question.difficulty.includes(difficulty);
+            });
+        }
 
-    if (questionType) {
-        let filter = questions.filter((question: Question) => {
-            return question.tags.includes(questionType);
-        });
+        if (!!hideCorrect) {
+            filter = filter.filter((question: Question, index: number) => {
+                return question.correct === hideCorrect;
+            });
+        }
```

Comment:
Instead of doing these separate, you could combine:

```
filter = filter.filter((question: Question, index: number) => {
                return question.tags.includes(type) && 
                     question.answer
                    .toLowerCase()
                    .includes(name.toLocaleLowerCase()) &&
                   question.difficulty.includes(difficulty) &&
...
            });
```

Reply:
filter = filter.filter((question: Question, index: number) => {
                return isQuestionType(question, type) &&
                           isQuestionName(question, name) &&
...

            });

Reply:
define filter functions in a utils file